### PR TITLE
HAAR-1033 spring boot 3 upgrade

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/config/ResourceServerConfiguration.kt [hunk 1]
```diff
@@ -1,29 +1,41 @@
...
+          "/favicon.ico",
+          "/csrf",
+          "/health/**",
+          "/info",
+          "/h2-console/**",
+          "/v3/api-docs/**",
+          "/swagger-ui/**",
+          "/swagger-ui.html",
+        ).forEach { authorize(it, permitAll) }
+        authorize(anyRequest, authenticated)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/config/ResourceServerConfiguration.kt:35]: I would double-check this is everything (swagger stuff) as I'm sure the initial script adds additional lines in for swagger config files

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/Caseload.kt [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
-import javax.persistence.Table
+import org.hibernate.type.YesNoConverter
 
 const val DPS_CASELOAD = "NWEB"
 const val CENTRAL_ADMIN_CASELOAD = "CADM_I"
 
 @Entity
 @Table(name = "CASELOADS")
-@BatchSize(size = 100)
 data class Caseload(
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/Caseload.kt:19]: Is this no longer needed?

Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/Caseload.kt:19]: @SimonMitchellMOJ - please reply to this as I'm not sure

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/config/ResourceServerConfiguration.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/config/ResourceServerConfiguration.kt:35]: @SimonMitchellMOJ - do you know whether there is anything missing here?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/Caseload.kt (hunk 2 above)
Comment (SimonMitchellMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/Caseload.kt:19]: I think it might still be needed was added as part of HAAR-470

Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/Caseload.kt]: Re-instated

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**